### DISCUSSION TITLE
semantic-release for docker configuration

### DISCUSSION THREAD

**Discussion by Techrese** [semantic-release/semantic-release#2773] [Q&A] [answered]
Hi, kinda stuck for semantic-release with microservices when using gitlab and .net6

So the semantic-release is working perfectly however in microservices architecture, semantic-versions are also splitted amongst there services for example:

initial commit: semantic-release 1.0.0
add feat branch for service: semantic-release 1.1.0
add feat branch for service 2: semantic-release 1.2.0

From a developer perspective, since I have added 2 services both microservices are version 1.0.0
When adding a new api to service 2: 
semantic-release created 1.3.0 while from a developers perspective:
container 1: 1.0.0 (untouched) and container 2: would be 1.1.0

Currently my release stage/jobs does a simple docker-compose build and then push rather than having a job for each service and do a build/push

How would I get the behaviour like it should be splitted amongst each project?

I rather feel that semantic-release is good if you have 1 project or a global release to clients.

**Comment by travi** [comment, marked as answer]
semantic-release does not support monorepos. whether versioning libraries or services, this project expects a repository to be 1:1 with the versioned asset. what it sounds like you are wanting to accomplish is not possible with the core project

**Reply by Techrese** [reply]
hi, thanks for an answer, the repo is not a monorepo, i'm using branches, but yes microservices essentially is different, on the docs page I have read about a docker plugin, but I don't think this fits my answer aswell.

**Comment by piqjen-1xofFi-sikxyh** [comment]
Is there something I’m supposed to do I’m guessing to fix this? My name is Jennifer this is all new to me

H.B.O. Ministries Worldwide
Jesus loves u
